### DISCUSSION TITLE
Need use response_model_exclude with Advanced exclude

### DISCUSSION THREAD

**Discussion by xl9211** [fastapi/fastapi#9087] [Questions] [answered]
When i use like this: 

```Python
response_model_exclude = ={
    'data': {
        'departments': {'__all__': {'tp_id'}}
    }
}
```

But i found that dict type will force change to set.

```Python
def jsonable_encoder(
    obj: Any,
    include: Union[SetIntStr, DictIntStrAny] = None,
    exclude: Union[SetIntStr, DictIntStrAny] = set(),
    by_alias: bool = True,
    skip_defaults: bool = None,
    exclude_unset: bool = False,
    exclude_defaults: bool = False,
    exclude_none: bool = False,
    custom_encoder: dict = {},
    sqlalchemy_safe: bool = True,
) -> Any:
    ...
    if exclude is not None and not isinstance(exclude, set):
        exclude = set(exclude)
    ...
```

then exclude will be change to 

```Python
exclude = {'data'}
```

Could you let exclude and include support advanced usecase like this https://pydantic-docs.helpmanual.io/usage/exporting_models/#advanced-include-and-exclude ?

**Comment by usrbinsam** [comment]
+1 for nested model support in `response_model_exclude` and `response_model_include`

**Comment by Dawoodkhorsandi** [comment, marked as answer]
Advanced exclude is now supported by fastAPI.

**Comment by pspiagicw** [comment]
Hi, can somebody point to the documentation regarding this, feature update. Or maybe somebody can include it in the documentation if the feature is confirmed.
